packages/flutter_tools/lib/src/build_system/targets/darwin.dart

```diff
@@ -152,3 +154,96 @@ abstract class CheckDevDependencies extends Target {
     globals.stdio.stderrWrite('error: $message');
   }
 }
+
+abstract class UnpackDarwin extends Target {
+  const UnpackDarwin();
+
+  Future<void> copyFramework(
+    Environment environment, {
+    EnvironmentType? environmentType,
+    TargetPlatform? targetPlatform,
+    required Artifact framework,
+    required BuildMode buildMode,
+  }) async {
+    final String basePath = environment.artifacts.getArtifactPath(
+      framework,
+      platform: targetPlatform,
+      mode: buildMode,
+      environmentType: environmentType,
+    );
+
+    final ProcessResult result = await environment.processManager.run(<String>[
+      'rsync',
+      '-av',
+      '--delete',
+      '--filter',
+      '- .DS_Store/',
+      '--chmod=Du=rwx,Dgo=rx,Fu=rw,Fgo=r',
+      basePath,
+      environment.outputDir.path,
+    ]);
+    if (result.exitCode != 0) {
+      throw Exception(
+        'Failed to copy framework (exit ${result.exitCode}:\n'
+        '${result.stdout}\n---\n${result.stderr}',
+      );
+    }
+  }
+
+  /// Destructively thin Flutter.framework to include only the specified architectures.
```

Comment:
The allowed values of the `archs` argument isn't immediately obvious from this method. Consider adding a comment here of an expected value, like `"x86_64 arm64"`.

Alternatively, consider accepting a `List<DarwinArch>` instead.

Reply:
Since the string value comes from Xcode, I added a comment rather than parsing it into a list.